src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/LocalIdKeyIdentifierClause.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Globalization;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class LocalIdKeyIdentifierClause : SecurityKeyIdentifierClause
+    {
+        readonly string localId;
+        readonly Type[] ownerTypes;
+
+        public LocalIdKeyIdentifierClause(string localId)
+            : this(localId, (Type[])null)
+        {
+        }
+
+        public LocalIdKeyIdentifierClause(string localId, Type ownerType)
+            : this(localId, ownerType == null ? (Type[])null : new Type[] { ownerType })
+        {
+        }
+
+        public LocalIdKeyIdentifierClause(string localId, byte[] derivationNonce, int derivationLength, Type ownerType)
+            : this(null, derivationNonce, derivationLength, ownerType == null ? (Type[])null : new Type[] { ownerType })
+        {
+        }
+
+        internal LocalIdKeyIdentifierClause(string localId, Type[] ownerTypes)
+            : this(localId, null, 0, ownerTypes)
+        {
+        }
+
+        internal LocalIdKeyIdentifierClause(string localId, byte[] derivationNonce, int derivationLength, Type[] ownerTypes)
+            : base(null, derivationNonce, derivationLength)
+        {
+            if (localId == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("localId");
+            }
+            if (localId == string.Empty)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgument(SR.Format("ID can't be empty."));
```

Comment:
I've created a gist with all the IdentityModel string resources in them. You can find it [here](https://gist.github.com/mconnew/6adcae48fd5914140befe272289b60a1). The name of this one was LocalIdCannotBeEmpty and can be found [here](https://gist.github.com/mconnew/6adcae48fd5914140befe272289b60a1#file-resources-txt-L45). You don't need SR.Format with this one as there's no parameters.